### ISSUE TITLE
Tracking large size figures 

### ISSUE THREAD

**Issue by JosephLaiC** [workflowr/workflowr#282]
Dear author, I am trying to use `workflowr` to manage my project, and sometimes it creates large SVG files (over 100MB). I found it difficult to apply `git lfs track` to solve this issue because the workflowr check can only be done in `wflow_publish`. The only thing I could do is manually generating the graph with png format in the code chunk and inserting it into the Rmd at a specific location, but it makes me hard to track the versions of the figures. If it is possible for you to add a function in `workflowr` that can automatically add large files via `git lfs track`, that would be greatly appreciated.

**Comment by jdblischak** [comment]
@JosephLaiC Frustratingly this isn't possible. Workflowr uses [git2r](https://docs.ropensci.org/git2r), which in turns uses [libgit2](https://libgit2.org/), to execute Git commands. Surprisingly libgit2 still hasn't implemented convenient support for git-lfs.

As you can see in the linked Issues below, proper support for git-lfs has been a long-standing feature request for libgit2 (since 2015!). While libgit2 has some basic support for the filters used by git-lfs, I assume it must be non-trival to configure these to robustly support git-lfs since no one has done it yet. Neither of the 2 R clients (git2r, gert) have implemented support for git-lfs, nor have various other software tools that rely on libgit2. It's a real shame.

https://github.com/git-lfs/git-lfs/issues/375#issuecomment-204475919
https://github.com/cmarcusreid/git-status-cache/issues/18
https://github.com/libgit2/libgit2sharp/issues/1236
https://github.com/r-lib/gert/issues/160
https://github.com/ropensci/git2r/issues/433

Here's a reprex that we can use for future testing if libgit2 ever adds supports for git-lfs:

```sh
git lfs install
Rscript -e 'workflowr::wflow_start("test-lfs")'
cd test-lfs
git lfs track "*.svg"
git add .gitattributes
git commit -m "Track SVG files with git-lfs"
# confirm it works
echo {1..100} > test.svg
git add test.svg
git lfs status
git commit -m "test svg"
git lfs ls-files
# in workflowr
echo -e '\n```{r plot, dev=c("png", "svg")}\nplot(1:10)\n```\n' >> analysis/index.Rmd
Rscript -e 'workflowr::wflow_publish("analysis/index.Rmd")'
git lfs ls-files
git lfs status
git status
```

The file `test.svg` is properly tracked via git-lfs when I use `git add` at the command line. However, when using libgit2 via `wflow_publish()`, the svg files are ignored. Thus a potential workaround (though I appreciate this isn't convenient), is to manually add and commit the SVG files via Git at the command line.

```sh
# after having run wflow_publish()
find docs/ -name "*.svg" -type f | xargs git add
git commit -m "Commit SVG files"
```

Note that by exporting both png and svg files, you can still get the reproducibility feature of tracking past versions of the figures(ie with the PNG versions), and by manually using git/git-lfs, you can also keep SVG versions tracked by Git.

**Comment by JosephLaiC** [comment]
Thanks for your reply. The command `{r plot, dev="png", dpi=proper.size}` solved my issue immediately and also saved the sotage of LFS : ) . I also tried manually adding SVG files and it worked! Appreciate your help.

Actually I forgot to post my issues. When I applied `wflow_publish` and push the repository into github, I got the error
```
remote: error: GH001: Large files detected. You may want to try Git Large File Storage - https://git-lfs.github.com/.
```
This issue still exist after I run
```
git filter-branch -f --index-filter 'git rm --cached --ignore-unmatch /large/file/name'
```

The only thing I can do is re-pulling the past version of repository or maybe there are other ways to solve this issues I didn't know.

I think it is worth to add some warnings before workflowr generate big files and commit it : ).

**Comment by jdblischak** [comment]
> Thanks for your reply. The command {r plot, dev="png", dpi=proper.size} solved my issue immediately and also saved the sotage of LFS : ) . I also tried manually adding SVG files and it worked! Appreciate your help.

Glad I could help! Also, I remembered that we have an entire vignette on [Using large data files with workflowr](https://workflowr.github.io/workflowr/articles/wflow-10-data.html). It doesn't cover your exact use case (it assumes that the large data files are input, not output), but it might still be helpful to you.

> The only thing I can do is re-pulling the past version of repository or maybe there are other ways to solve this issues I didn't know.

If it's really important to you to keep the current history, I recommend [git-filter-repo](https://github.com/newren/git-filter-repo). It's fast, and super extensible. You can pass it custom Python snippets to perform the exact edits you want.

However, note that many workflowr features (like the table with links to past versions) assume you aren't rewriting history. Thus if you haven't made many changes since the commit that first introduced the large SVG files, it'd probably make the most sense to backup your data, do a `git reset`, and then copy-paste back the updates to the .Rmd files from the backup you made.

> I think it is worth to add some warnings before workflowr generate big files and commit it : ).

`wflow_git_commit()` should already be doing this. It should throw an error instead of committing large files. Maybe the fact that the large files were output image files allowed it to get passed this check. Could you please confirm that the commit that introduced the large files was performed by workflowr? You can check with `git log --stat`

https://github.com/workflowr/workflowr/blob/57891b1060c25e243a17a1bbd14a9cb49160a04b/R/wflow_git_commit.R#L117

https://github.com/workflowr/workflowr/blob/57891b1060c25e243a17a1bbd14a9cb49160a04b/R/wflow_git_commit.R#L255-L267

**Comment by JosephLaiC** [comment]
Run

```
Rscript -e 'workflowr::wflow_start("test-lfs")'
cd test-lfs
echo -e '\n```{r, dev=c("png", "svg")}\n'\
  'library(RColorBrewer)\n'\
  'my_matrix <- matrix(runif(5000*600), nrow=5000, ncol=600)\n'\
  'break_length <- 100\n'\
  'my_palette <- colorRampPalette(brewer.pal(9, "YlOrRd"))(break_length)\n'\
  'pheatmap::pheatmap(\n'\
  'mat           = my_matrix,\n'\
  'cluster_rows  = FALSE,\n'\
  'cluster_cols  = FALSE,\n'\
  'show_rownames = FALSE,\n'\
  'show_colnames = FALSE,\n'\
  'color         = my_palette,\n'\
  'legend_colors = my_palette)\n```' >> analysis/index.Rmd
Rscript -e 'workflowr::wflow_build("analysis/index.Rmd")'
ls -lhs docs/figure/index.Rmd/*
```

Result

```
1.8M -rw-rw-r-- 1 joseph joseph 1.7M Mar  1 18:23 docs/figure/index.Rmd/unnamed-chunk-1-1.png
567M -rw-rw-r-- 1 joseph joseph 646M Mar  1 18:23 docs/figure/index.Rmd/unnamed-chunk-1-1.svg
```

Run
```
Rscript -e 'workflowr::wflow_git_commit(".")'
```
Result looks like there was no checking 
```
Loading .Rprofile for the current workflowr project
This is workflowr version 1.7.0
Run ?workflowr for help getting started
Summary from wflow_git_commit

The following was run:

  $ git add .
  $ git commit -m "workflowr::wflow_git_commit(\".\")"

The following file(s) were included in commit 55d839e:
analysis/index.Rmd
docs/figure/index.Rmd/unnamed-chunk-1-1.png
docs/figure/index.Rmd/unnamed-chunk-1-1.svg
docs/index.html
docs/site_libs/bootstrap-3.3.5/
docs/site_libs/highlightjs-9.12.0/
docs/site_libs/jquery-3.6.0/
docs/site_libs/navigation-1.1/
```

Run

```
git log --stat
```

Result

```
commit 55d839ee1e74c232c9c99d510a817e7945b48b28 (HEAD -> master)
Author: JosephLaiC <joe5621624@gmail.com>
Date:   Wed Mar 1 18:26:37 2023 +0800

    workflowr::wflow_git_commit(".")

 analysis/index.Rmd                                                      |    15 +
 docs/figure/index.Rmd/unnamed-chunk-1-1.png                             |   Bin 0 -> 1782555 bytes
 docs/figure/index.Rmd/unnamed-chunk-1-1.svg                             |   Bin 0 -> 676498570 bytes
 docs/index.html                                                         |   586 +++++++
 docs/site_libs/bootstrap-3.3.5/css/bootstrap-theme.css                  |   587 +++++++
 docs/site_libs/bootstrap-3.3.5/css/bootstrap-theme.css.map              |     1 +
 docs/site_libs/bootstrap-3.3.5/css/bootstrap-theme.min.css              |     5 +
 docs/site_libs/bootstrap-3.3.5/css/bootstrap.css                        |  6799 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 docs/site_libs/bootstrap-3.3.5/css/bootstrap.css.map                    |     1 +
 docs/site_libs/bootstrap-3.3.5/css/bootstrap.min.css                    |     5 +
 docs/site_libs/bootstrap-3.3.5/css/cerulean.min.css                     |    11 +
 docs/site_libs/bootstrap-3.3.5/css/cosmo.min.css                        |    30 +
 docs/site_libs/bootstrap-3.3.5/css/darkly.min.css                       |    30 +
 docs/site_libs/bootstrap-3.3.5/css/flatly.min.css                       |    30 +
 docs/site_libs/bootstrap-3.3.5/css/fonts/Lato.ttf                       |   Bin 0 -> 81980 bytes
 docs/site_libs/bootstrap-3.3.5/css/fonts/LatoBold.ttf                   |   Bin 0 -> 82368 bytes
 docs/site_libs/bootstrap-3.3.5/css/fonts/LatoItalic.ttf                 |   Bin 0 -> 81332 bytes
 docs/site_libs/bootstrap-3.3.5/css/fonts/NewsCycle.ttf                  |   Bin 0 -> 28716 bytes
 docs/site_libs/bootstrap-3.3.5/css/fonts/NewsCycleBold.ttf              |   Bin 0 -> 28724 bytes
 docs/site_libs/bootstrap-3.3.5/css/fonts/OpenSans.ttf                   |   Bin 0 -> 34112 bytes
 docs/site_libs/bootstrap-3.3.5/css/fonts/OpenSansBold.ttf               |   Bin 0 -> 35760 bytes
 docs/site_libs/bootstrap-3.3.5/css/fonts/OpenSansBoldItalic.ttf         |   Bin 0 -> 33064 bytes
 docs/site_libs/bootstrap-3.3.5/css/fonts/OpenSansItalic.ttf             |   Bin 0 -> 32808 bytes
 docs/site_libs/bootstrap-3.3.5/css/fonts/OpenSansLight.ttf              |   Bin 0 -> 35340 bytes
 docs/site_libs/bootstrap-3.3.5/css/fonts/OpenSansLightItalic.ttf        |   Bin 0 -> 32680 bytes
 docs/site_libs/bootstrap-3.3.5/css/fonts/Raleway.ttf                    |   Bin 0 -> 63796 bytes
 docs/site_libs/bootstrap-3.3.5/css/fonts/RalewayBold.ttf                |   Bin 0 -> 62224 bytes
 docs/site_libs/bootstrap-3.3.5/css/fonts/Roboto.ttf                     |   Bin 0 -> 32652 bytes
 docs/site_libs/bootstrap-3.3.5/css/fonts/RobotoBold.ttf                 |   Bin 0 -> 32500 bytes
 docs/site_libs/bootstrap-3.3.5/css/fonts/RobotoLight.ttf                |   Bin 0 -> 32664 bytes
 docs/site_libs/bootstrap-3.3.5/css/fonts/RobotoMedium.ttf               |   Bin 0 -> 32580 bytes
 docs/site_libs/bootstrap-3.3.5/css/fonts/SourceSansPro.ttf              |   Bin 0 -> 35064 bytes
 docs/site_libs/bootstrap-3.3.5/css/fonts/SourceSansProBold.ttf          |   Bin 0 -> 34908 bytes
 docs/site_libs/bootstrap-3.3.5/css/fonts/SourceSansProItalic.ttf        |   Bin 0 -> 33864 bytes
 docs/site_libs/bootstrap-3.3.5/css/fonts/SourceSansProLight.ttf         |   Bin 0 -> 35368 bytes
 docs/site_libs/bootstrap-3.3.5/css/fonts/Ubuntu.ttf                     |   Bin 0 -> 73608 bytes
 docs/site_libs/bootstrap-3.3.5/css/journal.min.css                      |    24 +
 docs/site_libs/bootstrap-3.3.5/css/lumen.min.css                        |    37 +
 docs/site_libs/bootstrap-3.3.5/css/paper.min.css                        |    36 +
 docs/site_libs/bootstrap-3.3.5/css/readable.min.css                     |    24 +
 docs/site_libs/bootstrap-3.3.5/css/sandstone.min.css                    |    24 +
 docs/site_libs/bootstrap-3.3.5/css/simplex.min.css                      |    24 +
 docs/site_libs/bootstrap-3.3.5/css/spacelab.min.css                     |    36 +
 docs/site_libs/bootstrap-3.3.5/css/united.min.css                       |    18 +
 docs/site_libs/bootstrap-3.3.5/css/yeti.min.css                         |    50 +
 docs/site_libs/bootstrap-3.3.5/fonts/glyphicons-halflings-regular.eot   |   Bin 0 -> 20127 bytes
 docs/site_libs/bootstrap-3.3.5/fonts/glyphicons-halflings-regular.svg   |   288 ++++
 docs/site_libs/bootstrap-3.3.5/fonts/glyphicons-halflings-regular.ttf   |   Bin 0 -> 45404 bytes
 docs/site_libs/bootstrap-3.3.5/fonts/glyphicons-halflings-regular.woff  |   Bin 0 -> 23424 bytes
 docs/site_libs/bootstrap-3.3.5/fonts/glyphicons-halflings-regular.woff2 |   Bin 0 -> 18028 bytes
 docs/site_libs/bootstrap-3.3.5/js/bootstrap.js                          |  2363 ++++++++++++++++++++++++++++
 docs/site_libs/bootstrap-3.3.5/js/bootstrap.min.js                      |     7 +
 docs/site_libs/bootstrap-3.3.5/js/npm.js                                |    13 +
 docs/site_libs/bootstrap-3.3.5/shim/html5shiv.min.js                    |     7 +
 docs/site_libs/bootstrap-3.3.5/shim/respond.min.js                      |     8 +
 docs/site_libs/highlightjs-9.12.0/default.css                           |    21 +
 docs/site_libs/highlightjs-9.12.0/highlight.js                          |     2 +
 docs/site_libs/highlightjs-9.12.0/textmate.css                          |    19 +
 docs/site_libs/jquery-3.6.0/jquery-3.6.0.js                             | 10881 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 docs/site_libs/jquery-3.6.0/jquery-3.6.0.min.js                         |     2 +
 docs/site_libs/jquery-3.6.0/jquery-3.6.0.min.map                        |     1 +
 docs/site_libs/navigation-1.1/codefolding-lua.css                       |     9 +
 docs/site_libs/navigation-1.1/codefolding.js                            |    76 +
 docs/site_libs/navigation-1.1/sourceembed.js                            |    12 +
 docs/site_libs/navigation-1.1/tabsets.js                                |   141 ++
 65 files changed, 22223 insertions(+)

commit 6e7b0029399e49179109a67b3a7a186679d65d78
Author: JosephLaiC <joe5621624@gmail.com>
Date:   Wed Mar 1 18:23:47 2023 +0800

    Start workflowr project.

 .Rprofile            |  8 ++++++++
 .gitattributes       |  3 +++
 .gitignore           | 11 +++++++++++
 README.md            |  5 +++++
 _workflowr.yml       |  8 ++++++++
 analysis/_site.yml   | 17 +++++++++++++++++
 analysis/about.Rmd   | 10 ++++++++++
 analysis/index.Rmd   | 11 +++++++++++
 analysis/license.Rmd | 21 +++++++++++++++++++++
 code/README.md       |  3 +++
 data/README.md       |  3 +++
 docs/.nojekyll       |  0
 output/README.md     |  3 +++
 test-lfs.Rproj       | 16 ++++++++++++++++
 14 files changed, 119 insertions(+)
```

**Comment by jdblischak** [comment]
@JosephLaiC Thanks for the very thorough reprex! That made it very easy to pinpoint the problem

> `Rscript -e 'workflowr::wflow_git_commit(".")'`

This is the problem. Passing `.` is a Git-specific trick to specify "all files". Workflowr expands file globs using `Sys.glob()`, but `.` isn't a valid glob.

If you pass the image files directly with `wflow_git_commit("docs/figure/index.Rmd/*")`, then it properly prevents them from being committed. However, in general, it's not recommended to directly use `wflow_git_commit()` on anything in `docs/`. The recommended usage is to always use `wflow_publish()`, which commits the files in `docs/` only after re-executing the code. Unfortunately for this use case, `wflow_publish()` adds the directory `docs/figure/index.Rmd` (another Git trick to specify all the files in the directory), so again the R code can't check the individual files.


```sh
cd ..
rm -rf test-lfs
Rscript -e 'workflowr::wflow_start("test-lfs")'
cd test-lfs
echo -e '\n```{r, dev=c("png", "svg")}\n'\
  'library(RColorBrewer)\n'\
  'my_matrix <- matrix(runif(5000*600), nrow=5000, ncol=600)\n'\
  'break_length <- 100\n'\
  'my_palette <- colorRampPalette(brewer.pal(9, "YlOrRd"))(break_length)\n'\
  'pheatmap::pheatmap(\n'\
  'mat           = my_matrix,\n'\
  'cluster_rows  = FALSE,\n'\
  'cluster_cols  = FALSE,\n'\
  'show_rownames = FALSE,\n'\
  'show_colnames = FALSE,\n'\
  'color         = my_palette,\n'\
  'legend_colors = my_palette)\n```' >> analysis/index.Rmd
Rscript -e 'workflowr::wflow_build("analysis/index.Rmd")'
ls -lhs docs/figure/index.Rmd/*
Rscript -e 'workflowr::wflow_git_commit("docs/figure/index.Rmd/*")'
## Loading .Rprofile for the current workflowr project
## This is workflowr version 1.7.0
## Run ?workflowr for help getting started
## Error in check_file_sizes(files) :
##   All files to be committed must be less than 100 MB. This is the max
## file size able to be pushed to GitHub, and is in general a good
## practice to follow no matter what Git hosting service you are using.
## Large files will make each push and pull take much longer and increase
## the risk of the download timing out. Run Git directly in the Terminal
## if you really want to commit these files.
## Calls: <Anonymous> -> check_file_sizes
## Execution halted
Rscript -e 'workflowr::wflow_publish("analysis/index.Rmd")'
## Loading .Rprofile for the current workflowr project
## This is workflowr version 1.7.0
## Run ?workflowr for help getting started
## Current working directory: /home/jdblischak/test-lfs
## Building 1 file(s):
## Building analysis/index.Rmd
## log directory created: /tmp/RtmpQbqtit/workflowr
## Summary from wflow_publish
## 
## **Step 1: Commit analysis files**
## 
## Summary from wflow_git_commit
## 
## The following was run:
## 
##   $ git add analysis/index.Rmd
##   $ git commit -m "workflowr::wflow_publish(\"analysis/index.Rmd\")"
## 
## The following file(s) were included in commit c924286:
## analysis/index.Rmd
## 
## **Step 2: Build HTML files**
## 
## Summary from wflow_build
## 
## Settings:
##  combine: "or" clean_fig_files: TRUE
## 
## The following were built externally each in their own fresh R session:
## 
## docs/index.html
## 
## Log files saved in /tmp/RtmpQbqtit/workflowr
## 
## **Step 3: Commit HTML files**
## 
## Summary from wflow_git_commit
## 
## The following was run:
## 
##   $ git add docs/index.html docs/figure/index.Rmd docs/site_libs docs/.nojekyll
##   $ git commit -m "Build site."
## 
## The following file(s) were included in commit 9e65d79:
## docs/figure/index.Rmd/unnamed-chunk-1-1.png
## docs/figure/index.Rmd/unnamed-chunk-1-1.svg
## docs/index.html
## docs/site_libs/bootstrap-3.3.5/
## docs/site_libs/header-attrs-2.20/
## docs/site_libs/highlightjs-9.12.0/
## docs/site_libs/jquery-3.6.0/
## docs/site_libs/navigation-1.1/
```

Honestly, I'm not sure how to best handle this. The vignette [Using large data files with workflowr](https://workflowr.github.io/workflowr/articles/wflow-10-data.html) assumes the large files are inputs to the `.Rmd` files, not output saved in `docs/`. If I updated `wflow_publish()` to pass `docs/figure/index.Rmd/*` instead of `docs/figure/index.Rmd`, then you wouldn't be able to use `wflow_publish()` because it would always fail. And I don't want to switch it to a warning. Trying to undo history is an advanced Git topic, so I'd prefer to prevent the problem altogether. Thus my recommendation to you is to add `*.svg` to `.gitignore` so that workflowr ignores SVG files. Then you can manually force-add and commit the SVG files with git-lfs. Below is an example:

```sh
cd ..
rm -rf test-lfs
Rscript -e 'workflowr::wflow_start("test-lfs")'
cd test-lfs
echo -e '\n```{r, dev=c("png", "svg")}\n'\
  'library(RColorBrewer)\n'\
  'my_matrix <- matrix(runif(5000*600), nrow=5000, ncol=600)\n'\
  'break_length <- 100\n'\
  'my_palette <- colorRampPalette(brewer.pal(9, "YlOrRd"))(break_length)\n'\
  'pheatmap::pheatmap(\n'\
  'mat           = my_matrix,\n'\
  'cluster_rows  = FALSE,\n'\
  'cluster_cols  = FALSE,\n'\
  'show_rownames = FALSE,\n'\
  'show_colnames = FALSE,\n'\
  'color         = my_palette,\n'\
  'legend_colors = my_palette)\n```' >> analysis/index.Rmd
echo '*.svg' >> .gitignore
git add .gitignore
git commit -m "Require .svg files to be force added with -f"
git lfs track "*.svg"
git add .gitattributes
git commit -m "Track SVG files with git-lfs"
Rscript -e 'workflowr::wflow_publish("analysis/index.Rmd")'
git add -f docs/figure/index.Rmd/unnamed-chunk-1-1.svg
git commit -m "Manually commit SVG file"
git log --stat
## commit 51675c81cf919afbeddec1006a62935264404c39
## Author: John Blischak <jdblischak@gmail.com>
## Date:   Thu Mar 2 14:50:37 2023 -0500
## 
##     Manually commit SVG file
## 
##  docs/figure/index.Rmd/unnamed-chunk-1-1.svg | 3 +++
##  1 file changed, 3 insertions(+)
## 
## commit 7f6a567426028b8fbed4b662616e2a923d08e16a
## Author: John Blischak <jdblischak@gmail.com>
## Date:   Thu Mar 2 14:50:31 2023 -0500
## 
##     Build site.
## 
##  docs/figure/index.Rmd/unnamed-chunk-1-1.png        |   Bin 0 -> 1782504 bytes
##  docs/index.html                                    |   604 +
##  .../bootstrap-3.3.5/css/bootstrap-theme.css        |   587 +
##  .../bootstrap-3.3.5/css/bootstrap-theme.css.map    |     1 +
##  .../bootstrap-3.3.5/css/bootstrap-theme.min.css    |     5 +
##  docs/site_libs/bootstrap-3.3.5/css/bootstrap.css   |  6799 ++++++++++++
##  .../bootstrap-3.3.5/css/bootstrap.css.map          |     1 +
##  .../bootstrap-3.3.5/css/bootstrap.min.css          |     5 +
##  .../site_libs/bootstrap-3.3.5/css/cerulean.min.css |    11 +
##  docs/site_libs/bootstrap-3.3.5/css/cosmo.min.css   |    30 +
##  docs/site_libs/bootstrap-3.3.5/css/darkly.min.css  |    30 +
##  docs/site_libs/bootstrap-3.3.5/css/flatly.min.css  |    30 +
##  docs/site_libs/bootstrap-3.3.5/css/fonts/Lato.ttf  |   Bin 0 -> 81980 bytes
##  .../bootstrap-3.3.5/css/fonts/LatoBold.ttf         |   Bin 0 -> 82368 bytes
##  .../bootstrap-3.3.5/css/fonts/LatoItalic.ttf       |   Bin 0 -> 81332 bytes
##  .../bootstrap-3.3.5/css/fonts/NewsCycle.ttf        |   Bin 0 -> 28716 bytes
##  .../bootstrap-3.3.5/css/fonts/NewsCycleBold.ttf    |   Bin 0 -> 28724 bytes
##  .../bootstrap-3.3.5/css/fonts/OpenSans.ttf         |   Bin 0 -> 34112 bytes
##  .../bootstrap-3.3.5/css/fonts/OpenSansBold.ttf     |   Bin 0 -> 35760 bytes
##  .../css/fonts/OpenSansBoldItalic.ttf               |   Bin 0 -> 33064 bytes
##  .../bootstrap-3.3.5/css/fonts/OpenSansItalic.ttf   |   Bin 0 -> 32808 bytes
##  .../bootstrap-3.3.5/css/fonts/OpenSansLight.ttf    |   Bin 0 -> 35340 bytes
##  .../css/fonts/OpenSansLightItalic.ttf              |   Bin 0 -> 32680 bytes
##  .../bootstrap-3.3.5/css/fonts/Raleway.ttf          |   Bin 0 -> 63796 bytes
##  .../bootstrap-3.3.5/css/fonts/RalewayBold.ttf      |   Bin 0 -> 62224 bytes
##  .../site_libs/bootstrap-3.3.5/css/fonts/Roboto.ttf |   Bin 0 -> 32652 bytes
##  .../bootstrap-3.3.5/css/fonts/RobotoBold.ttf       |   Bin 0 -> 32500 bytes
##  .../bootstrap-3.3.5/css/fonts/RobotoLight.ttf      |   Bin 0 -> 32664 bytes
##  .../bootstrap-3.3.5/css/fonts/RobotoMedium.ttf     |   Bin 0 -> 32580 bytes
##  .../bootstrap-3.3.5/css/fonts/SourceSansPro.ttf    |   Bin 0 -> 35064 bytes
##  .../css/fonts/SourceSansProBold.ttf                |   Bin 0 -> 34908 bytes
##  .../css/fonts/SourceSansProItalic.ttf              |   Bin 0 -> 33864 bytes
##  .../css/fonts/SourceSansProLight.ttf               |   Bin 0 -> 35368 bytes
##  .../site_libs/bootstrap-3.3.5/css/fonts/Ubuntu.ttf |   Bin 0 -> 73608 bytes
##  docs/site_libs/bootstrap-3.3.5/css/journal.min.css |    24 +
##  docs/site_libs/bootstrap-3.3.5/css/lumen.min.css   |    37 +
##  docs/site_libs/bootstrap-3.3.5/css/paper.min.css   |    36 +
##  .../site_libs/bootstrap-3.3.5/css/readable.min.css |    24 +
##  .../bootstrap-3.3.5/css/sandstone.min.css          |    24 +
##  docs/site_libs/bootstrap-3.3.5/css/simplex.min.css |    24 +
##  .../site_libs/bootstrap-3.3.5/css/spacelab.min.css |    36 +
##  docs/site_libs/bootstrap-3.3.5/css/united.min.css  |    18 +
##  docs/site_libs/bootstrap-3.3.5/css/yeti.min.css    |    50 +
##  .../fonts/glyphicons-halflings-regular.eot         |   Bin 0 -> 20127 bytes
##  .../fonts/glyphicons-halflings-regular.ttf         |   Bin 0 -> 45404 bytes
##  .../fonts/glyphicons-halflings-regular.woff        |   Bin 0 -> 23424 bytes
##  .../fonts/glyphicons-halflings-regular.woff2       |   Bin 0 -> 18028 bytes
##  docs/site_libs/bootstrap-3.3.5/js/bootstrap.js     |  2363 ++++
##  docs/site_libs/bootstrap-3.3.5/js/bootstrap.min.js |     7 +
##  docs/site_libs/bootstrap-3.3.5/js/npm.js           |    13 +
##  .../bootstrap-3.3.5/shim/html5shiv.min.js          |     7 +
##  docs/site_libs/bootstrap-3.3.5/shim/respond.min.js |     8 +
##  docs/site_libs/header-attrs-2.20/header-attrs.js   |    12 +
##  docs/site_libs/highlightjs-9.12.0/default.css      |    21 +
##  docs/site_libs/highlightjs-9.12.0/highlight.js     |     2 +
##  docs/site_libs/highlightjs-9.12.0/textmate.css     |    19 +
##  docs/site_libs/jquery-3.6.0/jquery-3.6.0.js        | 10881 +++++++++++++++++++
##  docs/site_libs/jquery-3.6.0/jquery-3.6.0.min.js    |     2 +
##  docs/site_libs/jquery-3.6.0/jquery-3.6.0.min.map   |     1 +
##  docs/site_libs/navigation-1.1/codefolding-lua.css  |     9 +
##  docs/site_libs/navigation-1.1/codefolding.js       |    76 +
##  docs/site_libs/navigation-1.1/sourceembed.js       |    12 +
##  docs/site_libs/navigation-1.1/tabsets.js           |   141 +
##  63 files changed, 21950 insertions(+)
## 
## commit 385b514a3c3db46a59b84420095a103826371166
## Author: John Blischak <jdblischak@gmail.com>
## Date:   Thu Mar 2 14:49:42 2023 -0500
## 
##     workflowr::wflow_publish("analysis/index.Rmd")
## 
##  analysis/index.Rmd | 15 +++++++++++++++
##  1 file changed, 15 insertions(+)
## 
## commit c186b8a8dc44d180258cb363c20054e7d1088668
## Author: John Blischak <jdblischak@gmail.com>
## Date:   Thu Mar 2 14:49:41 2023 -0500
## 
##     Track SVG files with git-lfs
## 
##  .gitattributes | 1 +
##  1 file changed, 1 insertion(+)
## 
## commit eb87d7fe172f254311c35675a03906532101180d
## Author: John Blischak <jdblischak@gmail.com>
## Date:   Thu Mar 2 14:49:41 2023 -0500
## 
##     Require .svg files to be force added with -f
## 
##  .gitignore | 1 +
##  1 file changed, 1 insertion(+)
## 
## commit 76f452a7727aa58f11e8aab2ae1cd1c529bc90d2
## Author: John Blischak <jdblischak@gmail.com>
## Date:   Thu Mar 2 14:49:41 2023 -0500
## 
##     Start workflowr project.
## 
##  .Rprofile            |  8 ++++++++
##  .gitattributes       |  3 +++
##  .gitignore           | 11 +++++++++++
##  README.md            |  5 +++++
##  _workflowr.yml       |  8 ++++++++
##  analysis/_site.yml   | 17 +++++++++++++++++
##  analysis/about.Rmd   | 10 ++++++++++
##  analysis/index.Rmd   | 11 +++++++++++
##  analysis/license.Rmd | 21 +++++++++++++++++++++
##  code/README.md       |  3 +++
##  data/README.md       |  3 +++
##  docs/.nojekyll       |  0
##  output/README.md     |  3 +++
##  test-lfs.Rproj       | 16 ++++++++++++++++
##  14 files changed, 119 insertions(+)
```
